packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1475,21 +1199,352 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
         key: const ValueKey<String>('time-picker-dial'),
         painter: _DialPainter(
           selectedValue: selectedDialValue,
-          primaryOuterLabels: primaryOuterLabels,
-          primaryInnerLabels: primaryInnerLabels,
-          secondaryOuterLabels: secondaryOuterLabels,
-          secondaryInnerLabels: secondaryInnerLabels,
+          primaryLabels: primaryLabels,
+          secondaryLabels: secondaryLabels,
           backgroundColor: backgroundColor,
-          accentColor: themeData.accentColor,
+          accentColor: accentColor,
+          dotColor: theme.colorScheme.surface,
           theta: _theta.value,
-          activeRing: _activeRing,
           textDirection: Directionality.of(context),
         ),
       ),
     );
   }
 }
 
+class _TimePickerInput extends StatefulWidget {
+  const _TimePickerInput({
+    Key key,
+    @required this.initialSelectedTime,
+    @required this.helpText,
+    @required this.onChanged,
+  }) : assert(initialSelectedTime != null),
+       super(key: key);
+
+  /// The time initially selected when the dialog is shown.
+  final TimeOfDay initialSelectedTime;
+
+  /// Optionally provide your own help text to the time picker.
+  final String helpText;
+
+  final ValueChanged<TimeOfDay> onChanged;
+
+  @override
+  _TimePickerInputState createState() => _TimePickerInputState();
+}
+
+class _TimePickerInputState extends State<_TimePickerInput> {
+  TimeOfDay _selectedTime;
+  bool hourHasError = false;
+  bool minuteHasError = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedTime = widget.initialSelectedTime;
+  }
+
+  int _parseHour(String value) {
+    if (value == null || value.isEmpty) {
```

Comment:
Don't need `|| value.isEmpty` because `int.tryParse('')` (which is done below) will return null

Reply:
true. done